lua/lsp_extensions/dart/closing_labels.lua

```diff
@@ -0,0 +1,69 @@
+--[[
+## Closing Labels
+
+** Method: 'dart/textDocument/publishClosingLabels**
+Direction: Server -> Client Params: { uri: string, labels: { label: string, range: Range }[] }
+This notifies the client when closing label information is available (or updated) for a file.
+
+Since this is a notification, the callback needs to be registered in the client's callbacks table.
+This can be achieved with nvim_lspconfig with this minimal config.
+```lua
+nvim_lsp.dartls.setup{
+  init_options = {
+    closingLabels = true,
+  },
+  callbacks = {
+    ['dart/textDocument/publishClosingLabels'] = require('lsp_extensions.dart.closing_labels').get_callback{},
+  },
+}
+```
+
+https://github.com/dart-lang/sdk/blob/master/pkg/analysis_server/tool/lsp_spec/README.md#darttextdocumentpublishclosinglabels-notification
+--]]
+
+local M = {}
+-- Stored labels from the server's notifications keyed by uri
+local all_labels = {}
+-- Namespace for the virtual text
+local closing_labels_ns = vim.api.nvim_create_namespace('lsp_extensions.dart.closing_labels')
+
+-- Gets a callback to register to the dartls publishClosingLabels notification.
+-- @tparam table a table of options: highlight, prefix
+M.get_callback = function(opts)
+
+  local get_draw_labels = function(opts)
+    return function()
+      opts = opts or {}
+      local highlight = opts.highlight or "Comment"
```

Comment:
These kind of things can be moved up one or more levels, so that they are only evaluated once, rather than every time we call the function (not a big deal for perf obviously, more just a matter of coding style)